### PR TITLE
[onert] Rename/Rearrange LowerInfo classes

### PR DESCRIPTION
**Pull request by wateret**
Rename LowerInfo related classes that were misleading.
Also they should be in `compiler` module.

- `ir::operand::LowerInfo` -> `compiler::OperandLowerInfo`
- `ir::operation::LowerInfo` -> `compiler::OpSequenceLowerInfo`
- `ir::operand::PermuteFactorSet` -> `compiler::PermuteFactorSet`
- `ir::operand::PermuteFactor` -> `compiler::PermuteFactor`
- `ir::LowerInfoMap` -> `compiler::GraphLowerInfo`

ONE-DCO-1.0-Signed-off-by: Hanjoung Lee <hanjoung.lee@samsung.com>

### CODE REVIEW

**Comment by ragmani** [comment]
@nnfw-bot test cross-runtime-test-debug

**Comment by ragmani** [comment]
@nnfw-bot test cross-runtime-test-release

File: runtime/onert/core/include/compiler/GraphLowerInfo.h [hunk 1]
```diff
@@ -14,29 +14,29 @@
  * limitations under the License.
  */
 
-#ifndef __ONERT_IR_LOWER_INFO_MAP_H__
-#define __ONERT_IR_LOWER_INFO_MAP_H__
+#ifndef __ONERT_GRAPH_LOWER_INFO_H__
```
Comment (hseok-oh) [inline runtime/onert/core/include/compiler/GraphLowerInfo.h:17]: (Optional) `__ONERT_GRAPH_LOWER_INFO_H__` -> `__ONERT_COMPILER_GRAPH_LOWER_INFO_H__`

**Review by hseok-oh — APPROVED** [review]
LGTM

**Review by glistening — APPROVED** [review]
LGTM

File: runtime/onert/core/include/compiler/GraphLowerInfo.h (hunk 1 above)
Comment (wateret) [reply runtime/onert/core/include/compiler/GraphLowerInfo.h:17]: I will make a fix after merging, thanks!

**Pull request merged**